### PR TITLE
HLM-6019 adding show error in upload preview screen  and navigation restrictions

### PR DESCRIPTION
**Pull request by siddhant-nawale-egov**

### CODE REVIEW

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/JsonPreviewInExcelForm.js [hunk 1]
```diff
@@ -45,8 +45,19 @@ export const JsonPreviewInExcelForm = (props) => {
...
+                    let headerName = sheetsData?.[currentSheetName]?.[0]?.[cellIndex]
+                    let error;
+                    if(headerName)
+                      error = props?.errorLocationObject?.[currentSheetName]?.[rowIndex]?.[headerName]
                     return (
-                      <td key={cellIndex} style={!Number.isNaN(parseFloat(rowData[cellIndex])) ? { textAlign: "end" } : {}}>
+                      <td
+                        key={cellIndex}
+                        style={{
+                          ...(!isNaN(rowData[cellIndex]) ? { textAlign: "end" } : {}),
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/JsonPreviewInExcelForm.js:56]: doesnt yield the same results. it give 2Wn as numaric

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js [hunk 2]
```diff
@@ -4,23 +4,20 @@ ajv.addKeyword("isRequired");
 ajv.addKeyword("isLocationDataColumns");
 ajv.addKeyword("isRuleConfigureInputs");
 ajv.addKeyword("isFilterPropertyOfMapSection");
-ajv.addKeyword("isVisualizationPropertyOfMapSection")
+ajv.addKeyword("isVisualizationPropertyOfMapSection");
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js:7]: they are getting assigned later on

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Nagivator.js [hunk 3]
```diff
@@ -43,16 +43,22 @@ const Navigator = memo((props) => {
...
+    if (checkDataCompletion === "invalid") {
+      if (navigationEvent && navigationEvent.name === "next") {
+        setToast(t("MICROPLAN_PLEASE_FILL_ALL_THE_FIELDS_AND_RESOLVE_ALL_THE_ERRORS"));
+      } else if (navigationEvent && navigationEvent.name === "step" && navigationEvent.step != undefined) {
+        if (navigationEvent.step > currentPage.id) setToast(t("MICROPLAN_PLEASE_FILL_ALL_THE_FIELDS_AND_RESOLVE_ALL_THE_ERRORS"));
+        else onStepClick(navigationEvent.step);
+      } else if (navigationEvent && navigationEvent.name === "previousStep") previousStep();
+      setCheckDataCompletion("false");
+    }
+  }, [checkDataCompletion]);
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Nagivator.js:56]: I want it to only run when checkDataCompletion changes

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/JsonPreviewInExcelForm.js [hunk 4]
```diff
@@ -45,8 +45,17 @@ export const JsonPreviewInExcelForm = (props) => {
               {sheetsData?.[currentSheetName]?.slice(1).map((rowData, rowIndex) => (
                 <tr key={rowIndex}>
                   {Object.values(sheetsData?.[currentSheetName]?.[0])?.map((_, cellIndex) => {
+                    const headerName = sheetsData?.[currentSheetName]?.[0]?.[cellIndex]
+                    const error = headerName ? props?.errorLocationObject?.[currentSheetName]?.[rowIndex]?.[headerName] : undefined;
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/JsonPreviewInExcelForm.js:49]: check again

[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/JsonPreviewInExcelForm.js:49]: i have them as const

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js:7]: Ok but they already are there

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**